packages/react-components/src/components/timegraph-output-component.tsx

```diff
@@ -384,7 +500,105 @@ export class TimegraphOutputComponent extends AbstractTreeOutputComponent<Timegr
         };
     }
 
+    private updateMarkersData(rangeEvents: TimelineChart.TimeGraphAnnotation[], newRange: TimelineChart.TimeGraphRange, newResolution: number) {
+        const annotationEntries: Entry[] = [];
+        const markers: Map<string, TimelineChart.TimeGraphState[]> = new Map();
+        const rows: TimelineChart.TimeGraphRowModel[] = [];
+        const filteredEvents = rangeEvents.filter(event => this.annotationMarkers?.includes(event.category));
```

Comment:
We might want to also filter the events that do not overlap with the current view range. The model is fetched for a wider range than the viewable range.

However this would need to be done when the view range changes, not when fetching the model... A row could appear and disappear as you browse in your cached data's range without ever updating the model.